cuebot/scheduler-sim/simulate.py

```
@@ -0,0 +1,2076 @@
+        # (instance 0 -- it starts first, so it holds the advisory lock) and
+        # prove the WHOLE service fails over: the standby must both BOOK new
+        # frames and ACCEPT new submissions after the kill. Every client
+        # re-dials the survivor via SIM_CUEBOT_GRPC_FALLBACKS (set above):
+        # fake_rqd/rqd_report for completions/status, and the feeder for
+        # LaunchSpec -- exactly like a real farm's multi-cuebot client config.
+        D = args.failover_test
+        half = D // 2
+        log(f"FAILOVER: load for {half}s, then killing the leader cuebot ...")
+        subprocess.run([VENV_PY, "live_stats.py", str(half), "5"], cwd=FARM)
+        try:
+            pid = int(open("/tmp/sim-cuebot-0.pid").read().strip())
```

Ruff (0.15.21)

[error] Probable insecure usage of temporary file or directory: "/tmp/sim-cuebot-0.pid"

(S108)